### PR TITLE
actions: add go-mod check

### PR DESCRIPTION
**Pull request by Rolinh**
This check ensures that vendored deps are "clean".

### CODE REVIEW

**Comment by Rolinh** [comment]
Blocked on #381

**Review by glibsm — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]
thanks @Rolinh for me `go mod tidy` was failing with this error:
```
% go mod tidy
go: google.golang.org/grpc@v1.33.0: reading google.golang.org/grpc/go.mod at revision v1.33.0: unknown revision v1.33.0
```
and i didn't catch the error.

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -32,3 +32,18 @@ jobs:
       with:
         name: release
         path: release
+  go-mod:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: '1.15.2'
```
Comment (michi-covalent) [inline .github/workflows/unittests.yml:41]: should we bump go version in go.mod as well? https://github.com/cilium/hubble/blob/f7d360b8921a5b958e07109983b8ca5e9cb40ab8/go.mod#L3

Comment (Rolinh) [reply .github/workflows/unittests.yml:41]: I don't think we have any reason to unless we want to use Go features that are not available in go 1.14.

**Pull request merged**